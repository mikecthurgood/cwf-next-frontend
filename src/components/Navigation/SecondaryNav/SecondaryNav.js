import React, { useContext } from 'react';
import Styled from 'styled-components';
// import { Dispatch, State } from '../../context/Store'
// import { Link } from 'react-router-dom';
// import AuthForm from '../form/AuthForm'

const SecondaryNav = ({slug, loginHandler, signupHandler}) => {

    // const { loginError, user, loginMenuVisible } = useContext(State)
    // const dispatch = useContext(Dispatch)
    function loginMenuToggle () {
        dispatch({type: 'setLoginMenuVisible', data: !loginMenuVisible})
    }

    return (
        !slug && (<NavBar>
            {/* <Link to="/"><img src={require('../../assets/clambr_logo_blue.png')} alt="clambr logo" id='main-logo' /></Link>
            {!user.username ?
            <>
                 <div className='login-dropdown'>
                    <AuthForm 
                    loginError={loginError} 
                    onSubmit={loginHandler} 
                    signup={false} 
                    user={user}
                    loginToggle={() => {}}
                    mobileMenuToggle={() => {}}
                    loginMenuToggle={() => {}}
                    loginHandler={loginHandler}
                    signupHandler={signupHandler}
                />
                 </div>
            </>
                :
                <div className="navbar-account-details"><p onClick={loginMenuToggle} className='nav-username'>Hi {user.username}!</p></div>
            }

            <div className={`login-menu ${user.username ? 'logged-in' : '' }`}>
                <img onClick={loginMenuToggle} src={require('../../assets/login-icon.jpg')} alt="" />
            </div> */}
        </NavBar >)
    )
}

export default SecondaryNav

const NavBar = Styled.div`
      position: fixed;
      top: 0;
      width: 100%;
      text-align: right;
      padding: 0;
      top: 50px;
      height: 40px;
      z-index: 100;
      background-color: #ED5656;
`


