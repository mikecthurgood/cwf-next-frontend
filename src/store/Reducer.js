const reducer = (state, action) => {
    switch (action.type) {
        
        case 'setFilterSelection':
            return { ...state, filterSelection: action.data}
        case 'setLoginError':
            return { ...state, loginError: action.data}
        case 'setLoginMenuVisible':
            return { ...state, loginMenuVisible: action.data}
        case 'setMobileMenuVisible':
            return { ...state, mobileMenuVisible: action.data}
        case 'setSearchBarVisibility':
            return { ...state, searchBarVisible: action.data}
        case 'setSearchFilter':
            return { ...state, searchFilter: action.data}
        case 'setScrollPosition':
            return { ...state, scrollPosition: action.data}
        case 'setSignUpSuccess':
            return { ...state, signUpSuccess: action.data}
        case 'setSingleWall':
            return { ...state, singleWall: action.data}
        case 'setSortInputVisibility':
            return { ...state, sortInputVisible: action.data}
        case 'setUser':
            return { ...state, user: action.data}
        case 'setPostCode':
            return { ...state, userPostCode: action.data}
        case 'setWalls':
            return { ...state, walls: action.data}
        case 'signOut' :
            localStorage.removeItem('token')
            localStorage.removeItem('userName')
            localStorage.removeItem('userId')
            return { 
                ...state,
                user: {username: null, userId: null, isAuth: false, token: ''},
                mobileMenuVisible: false,
                loginMenuVisible: false
            }
        default: 
            throw new Error('Invalid action type dispatched')
    }
}
export default reducer