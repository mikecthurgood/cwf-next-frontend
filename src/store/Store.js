import React, { useReducer } from  'react';
import reducer from './Reducer'

const initialState = {
    filterSelection: [],
    loginError: false,
    loginMenuVisible: false,
    mobileMenuVisible: false,
    searchBarVisible: false,
    scrollPosition: 0,
    signUpSuccess: false,
    singleWall: {},
    sortInputVisible: false,
    user: { username: null, userId: null, isAuth: false, token: '' },
    userPostCode: '',
    walls: [],
}

const State = React.createContext(initialState)
const Dispatch = React.createContext(null)

const Store = ({children}) => {
    
    const [state, dispatch] = useReducer(reducer, initialState)

    return (
        <Dispatch.Provider value={dispatch} >
            <State.Provider value={state} >
                {children}
            </State.Provider>
        </Dispatch.Provider>
    )
}


export { State, Dispatch, Store }

