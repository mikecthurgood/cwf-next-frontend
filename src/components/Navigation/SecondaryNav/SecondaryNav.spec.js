import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import { Dispatch } from '../../../store/Store';
import SecondaryNav from '.';

describe('Secondary Nav', () => {
    afterEach(cleanup);
    it('render the component', () => {
        const dispatch = jest.fn();
        const { container } = render(
            <Dispatch.Provider value={dispatch}>
                <SecondaryNav />
            </Dispatch.Provider>,
        );

        expect(container).toMatchSnapshot()
    });
});
