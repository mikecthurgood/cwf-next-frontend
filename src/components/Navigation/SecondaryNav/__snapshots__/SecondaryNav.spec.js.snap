// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secondary Nav render the component 1`] = `
<div>
  <div
    class="SecondaryNav__NavBar-sc-1joq3q2-0 bbXYtC"
  />
</div>
`;
